src/assets/js/main.js

```diff
@@ -79,20 +79,55 @@ function scrollSidebarIntoView() {
   }
 }
 
-function adjustToc() {
-  // Adjustments to the jekyll-toc TOC.
+/**
+ * Adjusts the behavior of the table of contents (TOC) on the page.
+ * 
+ * This function enables a "scrollspy" feature on the TOC, where the active link in the TOC is updated
+ * based on the currently visible section in the page.
+ * 
+ * Enables a "back to top" button in the TOC header.
+ * returns {void}
+ */
+function adjustTOC() {
 
   const tocId = '#site-toc--side';
 
   const tocHeader = document.querySelector(tocId + ' header');
 
   if (tocHeader) {
-    tocHeader.addEventListener('click', function (_) {
-      $('html, body').animate({scrollTop: 0}, 'fast');
+    //Shouldn't it be a "to top" button? 
```

Comment:
This kind of code `scrollToTop` is actioned on click of TOC header

In the below imagem, you will need to click on the header, or `Contents` to do this work.
![image](https://github.com/flutter/website/assets/15316327/82b391b8-f7d0-468a-b7b6-c76d9f5f76e5)

Shoud it be a ^ `to top` button?

Reply:
You are right it's not intuitive and it won't help on mobile where the TOC isn't visible lower down in mobile layouts. The ideal would be a floating button in the bottom right once users have scrolled down a little bit. We're tracking improved functionality in https://github.com/flutter/website/issues/1950 which would be nice to have if you are interested in working on that.

For now, you can remove these comments since we're tracking the improvement in that issue :)

Reply:
I can help with the issue #1950, taking advantage that I'm already work on that in this moment.

PS: I'll remove comments, and in the future I'll make them in a review.. looks more profissional =D